src/device/mapping.rs

```diff
@@ -122,22 +130,24 @@ impl<'a, T: Copy, R: Resources, F: Factory<R>> Drop for RW<'a, T, R, F> where
 /// A service trait with methods for mapping already implemented.
 /// To be used by device back ends.
 #[allow(missing_docs)]
-pub trait Builder<R: Resources> {
+pub trait Builder<'a, R: Resources> {
     type RawMapping: Raw;
 
-    fn map_readable<T: Copy>(&mut self, Self::RawMapping, usize)
+    fn map_readable<T: Copy>(&'a mut self, Self::RawMapping, usize)
                     -> Readable<T, R, Self>;
```

Comment:
You mean `Readable<'a, T, R, Self>`? If yes then shouldn't, rust inference lifetime from self.